bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TableUtils.java

```diff
@@ -94,10 +94,10 @@ public static Long hash(Object obj, Node parent) {
                 }
                 return result;
             } else {
-                return (long) obj.hashCode();
+                return (long) obj.toString().hashCode();
```

Comment:
Can you check what can go wrong if the type of the key is int|string?

Reply:
@rpjayasekara

Reply:
Will check

Reply:
This goes wrong. Gets the same hash for 0 and "0"
```
public function main() {
   table<record { readonly int|string? k; }> key(k) t = table[];
   t.add({k: "0"});
   boolean a = t.hasKey(0);
}
```